### PR TITLE
Np 45128 update period endpoint

### PR DESCRIPTION
**Pull request by truhacevkir**

### CODE REVIEW

File: docs/openapi.yaml [hunk 1]
```diff
@@ -85,6 +85,57 @@ paths:
             application/json:
               schema:
                 $ref: "#/components/schemas/Problem"
+    put:
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/NviPeriod'
```
Comment (anetteOlli) [inline docs/openapi.yaml:94]: Consider using the request body evaluators:
x-amazon-apigateway-request-validators

Comment (anetteOlli) [reply docs/openapi.yaml:94]: This gives you the perk of faster bad request responses. -> bad request response ~ 5ms
And less need for validating the request body inside the lambda. -> faster 201 response
And this ensures that your openapi spec is not just theoretical, it's part of the actual implementation.

**Review by anetteOlli — CHANGES_REQUESTED** [review]

File: docs/openapi.yaml (hunk 1 above)
Comment (anetteOlli) [reply docs/openapi.yaml:94]: aws request validation documentation: https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-request-validation-sample-api-swagger.html

Comment (truhacevkir) [reply docs/openapi.yaml:99]: Have added body required, but it seems that there is no string validation, only number value can be validated.

File: nvi-rest/src/main/java/no/sikt/nva/nvi/rest/UpdateNviPeriodHandler.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import nva.commons.apigateway.ApiGatewayHandler;
+import nva.commons.apigateway.RequestInfo;
+import nva.commons.apigateway.exceptions.ApiGatewayException;
+
+
+public class UpdateNviPeriodHandler extends ApiGatewayHandler<NviPeriodDto, NviPeriodDto> {
+
+    private final NviService nviService;
+
+    public UpdateNviPeriodHandler(NviService nviService) {
```
Comment (torarnet) [inline nvi-rest/src/main/java/no/sikt/nva/nvi/rest/UpdateNviPeriodHandler.java:23]: Does a handler need a no param constructor?

**Review by torarnet — APPROVED** [review]
See comment

File: nvi-rest/src/main/java/no/sikt/nva/nvi/rest/UpdateNviPeriodHandler.java (hunk 2 above)
Comment (truhacevkir) [reply nvi-rest/src/main/java/no/sikt/nva/nvi/rest/UpdateNviPeriodHandler.java:23]: It does, but for now we do not have defaultService to use. I add default constructor with null service

**Pull request merged**